packages/fxa-settings/src/pages/ResetPassword/CompleteResetPassword/index.tsx

```diff
@@ -84,44 +84,8 @@ const CompleteResetPassword = ({
       },
     });
 
-  /* When the user clicks the confirm password reset link from their email, we check
-   * the status of the link. If the link is valid, we check if a recovery key is enabled.
-   * If there is a recovery key, we navigate to the `account_recovery_confirm_key` page.
-   * If there isn't, we stay on this page and continue a regular password reset.
-   * If users clicked the link leading back to this page from `account_recovery_confirm_key`,
-   * we assume the user has lost the key and pass along a `lostRecoveryKey` flag
-   * so we don't perform the check and redirect again.
-   *
-   * If the link is -not- valid, we render link expired or link damaged.
-   *
-   * Additionally, the user can submit an invalid account recovery key and receive back
-   * an `accountResetToken`, then follow the link back to this page. In this case, we
-   * should _not_ check if the 'token' parameter is valid, since it will be invalid after
-   * this token is provided to us.
-   */
-  useEffect(() => {
-    const checkPasswordForgotToken = async (token: string) => {
-      try {
-        const isValid = await account.resetPasswordStatus(token);
-        if (isValid) {
-          setLinkStatus(LinkStatus.valid);
-          handleRecoveryKeyStatus();
-        } else {
-          setLinkStatus(LinkStatus.expired);
-        }
-      } catch (e) {
-        setLinkStatus(LinkStatus.damaged);
-      }
-    };
-
-    const handleRecoveryKeyStatus = async () => {
-      if (!location.state?.lostRecoveryKey) {
-        await checkForRecoveryKeyAndNavigate(linkModel.email);
-      }
-      renderCompleteResetPassword();
-    };
-
-    const checkForRecoveryKeyAndNavigate = async (email: string) => {
+  const checkForRecoveryKeyAndNavigate = useCallback(
```

Comment:
Did things just get moved around a bit in this file?  It looks like that in the diff but I'm not sure.

Reply:
Yes, moved some functions out of the `useEffect` which reduced the number of required dependencies.